kong/init.lua

```diff
@@ -29,6 +29,13 @@ require "resty.core"
 local constants = require "kong.constants"
 
 do
+  -- if we're running nginx -t then don't initialize
+  if os.getenv("KONG_NGINX_CONF_CHECK") then
+    return {
+      init = function()
+      end,
+    }
+  end
```

Comment:
This should go below the check for required shms, otherwise configs lacking the shm would error during the actual `start`, and yet configuration validation would report a success...